### DISCUSSION TITLE
Mark BusinessBase object IsNew=false

### DISCUSSION THREAD

**Discussion by CROprogrammer** [MarimerLLC/csla#3364] [Questions] [answered]
Is there any way to mark BusinessBase object's `IsNew` property to false?

**Comment by rockfordlhotka** [comment]
MarkOld?

**Reply by CROprogrammer** [reply]
I have the next method in my MVC controller:

```
    [HttpPost]
    [ValidateAntiForgeryToken]
    public async Task<IActionResult> Edit(Project project)
    {
      project.MarkAsOld();

      if (await SaveObjectAsync(project, true))
      {
        return RedirectToAction(nameof(Index));
      }

      return View(project);
    }
```

and in my async business rule I check that Project's code is unique, and when I'm editing project I disable project code editing. So the way I determine if I should check if project code already exists is so that I check if object is new or not. If the object is new, then I do not check my async rule.
But somehow my async rule is being called, when I submit my form, before`Edit` method where I mark project as old. Why is that? How can I avoid that business rules are being checked before I call Save method in my MVC Controller's method?

**Reply by rockfordlhotka** [reply, marked as answer]
The rule is probably being executed when the postback data is bound to the properties as the `Project` object is being created so it can be passed into the `Edit` method - so before any of the code in the `Edit` method is executed.

This occurs in `Csla.Web.Mvc.Shared.CslaModelBinder`:
https://github.com/MarimerLLC/csla/blob/91921c07ac5679f986147554b9f8ddf350bbe2c0/Source/Csla.Web.Mvc.Shared/CslaModelBinder.cs#L59

I don't see a way to prevent this behavior without altering `CslaModelBinder` - probably to _globally_ disable the automatic running of rules, which would _require_ that your code in the controller call `CheckRules` when and if you want the rules to run.
